doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -0,0 +1,399 @@
+            "hidden": true
+        }
+    ],
+```
+
+#### Hidden Profiles and the Open New Tab shortcuts
+
+Currently, there are keyboard shortcuts for "Open New Tab With Profile
+&lt;N&gt;". These shortcuts will open up the Nth profile in the new tab
+dropdown. Considering we're adding the ability to remove profiles from that
+list, but keep them in the overall list of profiles, we'll need to make sure
+that the handler for that event still opens the Nth _visible_ profile.
+
+### Dynamic Profiles
+
+Sometimes, we may want to auto-generate a profile on the user's behalf. Consider
+the case of WSL distros on their machine, or VMs running in Azure they may want
+to auto-connect to. These _dynamic_ profiles have a source that might be added
+or removed after the app is installed, and they will be different from user to
+user.
+
+Currently, these profiles are only generated when a user first launches the
+Terminal. If they already have a `profiles.json` file, then we won't run the
+auto-generation behavior. This is obviously not great - if any new types of
+dynamic profiles are added, then users that already have the Terminal installed
+won't get any of these dynamic profiles. Furthemore, if any of the sources of
+these dynamic profiles are removed, then the app won't auto-remove the
+associated profile.
+
+In the new model, with a combined defaults & user settings, how should these
+dynamic profiles work?
+
+I propose we add functionality to automatically search for these profile sources
+and add/remove them on _every_ Terminal launch. To make this functionality work
+appropriately, we'll need to introduce a constraint on dynamic profiles.
+
+**For any dynamic profiles, they must be able to be generated using a stable
+GUID**. For example, any time we try adding the "Ubuntu" profile, we must be
+able to generate the same GUID every time. This way, when a dynamic profile
+generator runs, it can check if that profile source already has a profile
+associated with it, and do nothing (as to not create many duplicate "Ubuntu"
+profiles, for example).
+
+When a dynamic profile generator runs, it will determine what new profiles need
+to be added to the user settings, and it can append those to the list of
+profiles. The generator will return some sort of result indicating that it wants
+a save operation to occur. The app will then save the `profiles.json` file,
+including these new profiles.
+
+If a dynamic profile generator has determined that a profile should no longer be
+used, we don't want to totally remove the profile from the file - perhaps
+there's some data the user would like from those settings. Instead, we'll simply
+mark that profile as `"hidden": true`. That way, the profile will remain in
+their settings, but will not appear in the list of profiles.
+
+Should a dynamic profile generator try to create a profile that's already
+`hidden`, it'll make sure to set `hidden` back to false, making the original

```

Comment:
Concerned about the layering here. A dynamic profile generator should not be able to generate a profile _after the user has suppressed it_.. because the user gets final say over what happens to that GUID

Reply:
Also, question, wouldn't generation take place _before_ overlaying user settings?

Reply:
Sure okay, I'll pull that bit. 

I'd think that the generation would occur _after_ the user settings are loaded, so it could see if a profile it wants to create has already been serialized and persisted.

After the dynamic profiles are generated and everything is validated, we'll persist any changes, including writing new dynamic profiles back to `profiles.json`

Reply:
But if we consider that "you only want to commit deltas over top of dynamic profiles" (per our discussion on Teams), that means we have a four-phase load:

1. load default json
2. load user settings
3. load dynamic profiles
4. load user settings that override dynamic profile settings

the aforementioned mental model of dynamics being a different type of _default_ reduces that to 3:

1. load defaults
2. load dynamics
3. load user overrides (dynamics become hidden, shown, they get their color schemes changed, etc.)

Reply:
(circling back on this with more details from the above discussion) 

The model I'm going for is:
1. Load defaults
2. Load user settings
3. Generate dynamic profiles (if necessary)

Reply:
So we did a lot of discussing of this in #1321, so I'm going to resolve this thread for now.